src/framework/theme/components/calendar/calendar-range-cells.ts

```diff
@@ -155,19 +216,149 @@ export class NbCalendarRangeYearCellComponent<D> implements NbCalendarCell<D, Nb
     this.select.emit(this.date);
   }
 
-  private smallerThanMin(): boolean {
+  protected smallerThanMin(): boolean {
     return this.date && this.min && this.dateService.compareDates(this.yearEnd(), this.min) < 0;
   }
 
-  private greaterThanMax(): boolean {
+  protected greaterThanMax(): boolean {
     return this.date && this.max && this.dateService.compareDates(this.yearStart(), this.max) > 0;
   }
 
-  private yearStart(): D {
+  protected yearStart(): D {
     return this.dateService.getYearStart(this.date);
   }
 
-  private yearEnd(): D {
+  protected yearEnd(): D {
     return this.dateService.getYearEnd(this.date);
   }
+
+  protected isInRange(date: D, { start, end }: NbCalendarRange<D>): boolean {
+    if (start && end) {
+      const cellYear = this.dateService.getYear(date);
+      const startYear = this.dateService.getYear(start);
+      const endYear = this.dateService.getYear(end);
+
+      return cellYear >= startYear && cellYear <= endYear;
+    }
+
+    return this.dateService.isSameYear(date, start);
+  }
+}
+
+@Component({
+  selector: 'nb-calendar-range-month-cell',
+  template: `
+    <div class="cell-content">
+      {{ month }}
+    </div>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class NbCalendarRangeMonthCellComponent<D> extends NbBaseCalendarRangeCell<D>
```

Comment:
Would you mind splitting this file? I believe we need to move each component to a separate file because this file has too many lines.

Reply:
Done in 29c251ec1c82fc4450ef4cf246936f815eb1550f